File: src/main/resources/public/app/service/foodService.js
angular.module('app').service('foodService', ['$http' ,'$state',
  function($http, $state){
	
	 // local var to hold end point name
    var foodEndPoint = "/food";

    /**
	 * Call to web service to return all Vendors 
	 */
    function getAllVendors() {
        return $http.get(foodEndPoint + '/vendors' );
      }
    /**
	 * Call to web service to return all Categories 
	 */
    function getAllCategory() {
    	return $http.get(foodEndPoint + '/catagory' );
    }
    /**
	 * Call to web service to create Favorite food item 
	 */
    function createFood(fooddata) {
    	 return $http.post(foodEndPoint + "/createfood", fooddata);
    }
    /**
	 * Call to web service to return all fav food details  
	 */
    function getAllFavFood() {
    	console.log("createFood foodservices.js" );
    	return $http.get(foodEndPoint + '/favfood' )
    }
    /**
     * Call to retrieve all information about fav Food with corresponding ID
     */
    function getFavFoodDetailsID(foodid) {
    	return $http.get(foodEndPoint + '/id/' + foodid);
    }
    /**
	 * Call to web service to update Favorite food item 
	 */
    function updateCurentFood(fooddata) {
    	 return $http.put(foodEndPoint + "/updatefood", fooddata);
    }
    return {
    	getAllVendors : getAllVendors,
    	getAllCategory : getAllCategory,
    	createFood : createFood,
    	getAllFavFood : getAllFavFood,
    	getFavFoodDetailsID : getFavFoodDetailsID,
    	updateCurentFood : updateCurentFood
    };
    
    
    
}]);

File: src/main/java/com/steffyfinalproject/springboot/services/FoodService.java
package com.steffyfinalproject.springboot.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.steffyfinalproject.springboot.dao.FoodDao;
import com.steffyfinalproject.springboot.entities.Catagory;
import com.steffyfinalproject.springboot.entities.FavFood;
import com.steffyfinalproject.springboot.entities.Vendor;

/**
 * Contains info about all Food Services
 *
 */
@Service
public class FoodService {

	private String regexp = "[a-zA-z\\s]*";

	@Autowired
	FoodDao foodDao;

	public void setFoodDao(FoodDao foodDao) {
		this.foodDao = foodDao;
	}

	/**
	 * validation before entering to database
	 * 
	 * @param animal
	 * @return
	 */
	public Boolean validation(FavFood favFood) {

		String name = favFood.getName();

		if (name.matches(regexp)) {
			System.out.println("sucess");
			return true;
		} else {
			return false;
		}
	}
	/**
	 * Service to add favorite food
	 * 
	 * @throws Exception
	 */
	public void addFood(FavFood favfood) {

		Boolean result = validation(favfood);

		if (result) {
			foodDao.add(favfood);
		}

	}
	/**
	 * Service to retrieve all vendors
	 */
	public List<Vendor> getVendors() {
		return foodDao.getAllVendors();
	}

	/**
	 * Service to retrieve all categories
	 */
	public List<Catagory> getCategory() {
		return foodDao.getAllCategory();
	}

	/**
	 * Service to get details about all favorite food
	 */
	public List<FavFood> getAllFavFood() {
		return foodDao.getFavFoodDetails();
	}

	/**
	 * Service to get details about favorite food with corresponding ID
	 */
	public FavFood getFoodById(Integer foodId) {
		return foodDao.getFavFoodByID(foodId);
	}

	/**
	 * Service to update details about favorite food
	 * 
	 * @throws Exception
	 */
	public void update(FavFood favfood) {

		Boolean result = validation(favfood);

		if (result) {
			foodDao.updateFood(favfood);
		}

	}
}
